### ISSUE TITLE
Unable to select "Edit Web Service Attributes" for Web Service References in CIMIdentitiesJSP

### ISSUE THREAD

**Issue by swathirise123** [epri-dev/CIM-Identities#2]
As mentioned in the manual, I have deleted web service references and recreated them using the WSDL URL by right-clicking the project and selecting New --> web service client -->select the option for WSDL URL -->paste the URL for the SendCIMIdentities or QueryCIMIdentities. I'm not able to select “Wsimport Options” and change the wsdl location to another address.
![image](https://user-images.githubusercontent.com/37460450/38216543-b8b08fce-3690-11e8-8c7e-d1d233e8d96d.png)

Without doing this, I just proceeded with next steps in the manual. I clicked on "Clean and Rebuild", and "Run" CIMIdentitiesJSP. Then I got a screen as shown below.
![image](https://user-images.githubusercontent.com/37460450/38216209-b9f7261e-368f-11e8-99fd-41ecaeb8b84f.png)

When I clicked on "Accept", I got an error as shown below.
![image](https://user-images.githubusercontent.com/37460450/38216263-e11e9074-368f-11e8-9ba3-4835adfee695.png)

**Comment by DanielRLowe** [comment]
I still haven't figured out why it is that "Edit Web Service Attributes" is grayed out for some people and not others.  When that happens, you can get around it by deleting the attached Web Service Clients and adding new ones using the WSDL address generated by the Web Service.  So the first thing you should do is grab your ?WSDL addresses.  Yours is likely `http://localhost:8080/CIMIdentities-w/SendCIMIdentities?WSDL` and `http://localhost:8080/CIMIdentities-w/QueryCIMIdentities?WSDL`.  Note that this won't work unless you've deployed the Web Service (CIMIdentities-w).  To add the web service clients, right click the project and select "Add Web Service Client".  You might have to select "other" to find it, as the shortcut for it probably won't show up on the shortlist initially.

From there, if you're still having that same message come up, check your Glassfish log.  A common cause for that error is either the database isn't currently running, or the driver isn't being found (in which case I've found deleting the Postgres driver from your Library and adding it back in, then cleaning and rebuilding, tends to fix). 

Basically, if you recreate the Web Service clients and still have that message come up, share the Glassfish log so we can further pinpoint the actual issue(NetBeans should show it in the IDE once you hit cancel for the window saying that Netbeans Connector is debugging the browser).  Thanks!

Note:  At this point, if you reallllly wanted to, you could build your own web service client using whatever you wanted (VS Studio and ASP.NET for instance) by feeding them the WSDL addresses of the deployed web service.

**Comment by swathirise123** [comment]
Thank you for sharing more information. Now I am able to select "Edit Web Service Attributes". I clicked on "Clean and Rebuild", and "Run" CIMIdentitiesJSP. Then I got a screen as shown below.
![image](https://user-images.githubusercontent.com/37460450/38223324-fd9ef872-36af-11e8-8725-7ae8d1f01dc0.png)
When I clicked on "Accept", I still got an error as shown below.
![image](https://user-images.githubusercontent.com/37460450/38223343-14880a42-36b0-11e8-9f1c-232bd6c18955.png)

The problem seems to be with the Postgres JDBC driver. The glassfish server log from Netbeans IDE is attached here.
[Glassfish.log](https://github.com/epri-dev/CIM-Identities/files/1869804/Glassfish.log)

The file below consists of Domain Logs downloaded from http://localhost:4848
[server.log](https://github.com/epri-dev/CIM-Identities/files/1869851/server.log)
I followed the procedure correctly. Could you please help me why its not working?

**Comment by DanielRLowe** [comment]
Ah, your pictures help a lot.  So the JDBC Driver is actually only important for `CIMIdentities-w` since only the web service (not the client) directly talks to the database.  So reinstall the JDBC Driver to the Web Service library and rebuild/deploy.

Then when you create the JSPs, since `Edit Web Service Attribute` isn't showing up, you'll have to delete the clients (Just delete the whole Web Service References directory) and then add them both individually.  Once you do that you'll need to rebuild to create the auto-generated files.  Since you would add the web service clients in, you _might_ not need to edit the WSDLs for CIMIdentitiesJSP.  Check just in case and scroll to the bottom to make sure the localhost port matches up with your CIMIdentities-w.  

If it's still giving you the Null Pointer error message, I'll need to see your glassfish logs to help figure out what the problem really is.  You can find this in NetBeans on the same window (different tab) as the error message you pasted above.  

Sorry this isn't any easier.  This was meant more for another programmer to pick up and make additions to/modify than it was for a regular person to pick up and use, haha.  In hindsight, I could have made it that way too by creating a command script that would just launch the WAR file for you using Jetty or something.

![image](https://user-images.githubusercontent.com/15255368/38222563-782aa340-36b3-11e8-9c53-380a68468a1e.png)

**Comment by swathirise123** [comment]
You're right, I'm not a very good programmer. I'm deleted the whole zip file and downloaded new zip file. I repeated the whole process, still got same error.
![image](https://user-images.githubusercontent.com/37460450/38225432-840d5a20-36ba-11e8-84b4-a5e0970cdf36.png)

Netbeans IDE shows the following
![image](https://user-images.githubusercontent.com/37460450/38225372-3be55a4a-36ba-11e8-8b33-f6d4b45fc8b2.png)
Log files from Netbeans IDE 
[logs.txt](https://github.com/epri-dev/CIM-Identities/files/1869964/logs.txt)

**Comment by DanielRLowe** [comment]
> You're right, I'm not a very good programmer.

Don't worry, I'm not either :)  And that's not quite what I was saying haha.  What I was saying is that I could have simply made this easier in general so that no programming knowledge was needed at all, and then had the dirty back-end setup in addition to that.

Looking at your logs, it's definitely not getting picking up the driver correctly. I don't know what happened to make it suddenly not pick up the driver, but I added a failsafe to the code so that it'll grab the driver.  Download the latest version and let me know if that fixes the error (it worked on my end, but you know how these things go).  Thanks!

**Comment by swathirise123** [comment]
Finally I got the output with all your help. Thank you so much :). 
![image](https://user-images.githubusercontent.com/37460450/38230039-9888610e-36d1-11e8-901d-343d23d3f1f5.png)

**Comment by swathirise123** [comment]
When I inserted data, its working properly as shown below. When I tried to 'Modify'or 'Delete' data, It's not doing anything. (I can't see any update on the right side of the screen) 
![image](https://user-images.githubusercontent.com/37460450/38230688-bf282eae-36d4-11e8-992f-bc0d906786d5.png)

When I clicked on "View database" or "Query", its showing me an error as shown below. (I think its again the JDBC driver issue)
![image](https://user-images.githubusercontent.com/37460450/38230543-1002de24-36d4-11e8-83d9-4d55cc968150.png)

**Comment by DanielRLowe** [comment]
That's pretty strange.  Some files definitely went missing in the transition of making this public and I didn't test it afterwards because I assumed nothing was changed.  I've reverted the files back to how they were before that, so download the newest version and you should be able to set this up in less than 5 minutes.  You won't need to create the tables again.  Just clean and build CIMIdentities-w, deploy it, then clean and build either CIMIdentities-c or CIMIdentitiesJSP and you should have no issues running them.  (Localhost is set to 8080 already for all of these).

Closing this once you've confirmed it's working.  Sorry for the headache it's been!

**Comment by swathirise123** [comment]
It happens like that sometimes. But never mind, it's working perfectly for me right now. Thanks a lot for bearing my doubts.
![image](https://user-images.githubusercontent.com/37460450/38261917-47ed565e-3731-11e8-86e4-5f4fd053e83c.png)

![image](https://user-images.githubusercontent.com/37460450/38261833-084f1186-3731-11e8-924f-ffaebe117a0e.png)

![image](https://user-images.githubusercontent.com/37460450/38261842-0f5d828c-3731-11e8-9645-715c4f8829dd.png)
